rdbms/src/main/java/org/spine3/server/storage/rdbms/JdbcEntityStorage.java

```diff
@@ -0,0 +1,295 @@
+    }
+
+    private String setTableName(String sql) {
+        return String.format(sql, tableName);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @throws DatabaseException if an error occurs during an interaction with the DB
+     */
+    @Nullable
+    @Override
+    public M read(I id) throws DatabaseException {
+        final String idString = idToString(id);
+        try (ConnectionWrapper connection = dataSource.getConnection(true);
+             PreparedStatement statement = selectByIdStatement(connection, idString)) {
+            final M result = findById(statement);
+            return result;
+        } catch (SQLException e) {
+            logTransactionError(idString, e);
+            throw new DatabaseException(e);
+        }
+    }
+
+    private M findById(PreparedStatement statement) throws DatabaseException {
+        try (ResultSet resultSet = statement.executeQuery()) {
+            if (!resultSet.next()) {
+                return null;
+            }
+            final byte[] bytes = resultSet.getBytes(ENTITY);
+            final M message = toMessage(bytes);
+            return message;
+        } catch (SQLException e) {
+            throw new DatabaseException(e);
+        }
+    }
+
+    private M toMessage(byte[] bytes) {
+        final Any.Builder builder = Any.newBuilder();
+        builder.setTypeUrl(typeName.toTypeUrl());
+        builder.setValue(ByteString.copyFrom(bytes));
+        final M message = Messages.fromAny(builder.build());
+        return message;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @throws DatabaseException if an error occurs during an interaction with the DB
+     */
+    @Override
+    public void write(I id, M message) throws DatabaseException {
+        checkNotNull(id, "id");
+        checkNotNull(message, "message");
+
+        final String idString = idToString(id);
+        final byte[] serializedEntity = serialize(message);
+        try (ConnectionWrapper connection = dataSource.getConnection(false)) {
+            if (containsEntity(idString, connection)) {
+                update(idString, serializedEntity, connection);
+            } else {
+                insert(idString, serializedEntity, connection);
+            }
+        }
+    }
+
+    private byte[] serialize(M message) {
+        final Any any = toAny(message);
+        final byte[] bytes = any.getValue().toByteArray();
+        return bytes;
+    }
+
+    private void createTableIfDoesNotExist(String sql, String tableName) throws DatabaseException {
+        try (ConnectionWrapper connection = dataSource.getConnection(true);
+             PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.execute();
+        } catch (SQLException e) {
+            log().error("Error during table creation, table name = " + tableName, e);
+            throw new DatabaseException(e);
+        }
+    }
+
+    private boolean containsEntity(String id, ConnectionWrapper connection) {
```

Comment:
Have `connection` as the first parameter for all methods. We need to be consistent across the whole API.